### ISSUE TITLE
Unable to persist session after refresh

### ISSUE THREAD

**Issue by wombatdev** [lynndylanhurley/ng-token-auth#365]
Hi, I can't get user sessions to persist after refresh. Both user registration and user login are functioning, and I can navigate within the SPA in a session, but as soon as I refresh, I need to log in again. I'm using all of the default config for ng-token-auth and the user session controller is pretty basic:
```
function UserSessionsControllerFunction($scope, $state, $rootScope, $auth) {
        console.log("I'm in the user sessions controller!")
        var vm = this;

        vm.formSubmit = function(loginForm) {
            console.log($scope.loginForm);
            $auth.submitLogin($scope.loginForm)
                .then(function(resp) {
                    console.log("auth success");
                    console.log(resp);
            })
                .catch(function(resp) {
                    console.log("auth failure");
            });
        };

        $scope.$on('auth:login-error', function(ev, reason) {
            $scope.error = reason.errors[0];
        });

    }
```
Is there any additional setup I need to do with ipCookie? Or on the rails side?

Thanks!

**Comment by adonespitogo** [comment]
Have you found a solution to this @wombatdev ?

**Comment by adonespitogo** [comment]
Found out the `devise_auth_token` gem is not returning the right data (tokens, client_id, etch) upon successful  login. This causes session data `auth_headers` become empty object and session is not persisted after refresh.

**Comment by adonespitogo** [comment]
This fixed my problem. Turns out I added a new validation to my model which makes the user invalid https://github.com/lynndylanhurley/devise_token_auth/issues/721

**Comment by wombatdev** [comment]
@adonespitogo I have not gotten it working. My Rails headers were already set up correctly, so I don't think it's that.

I can log in, and if I include `$auth.validateUser` in the response, I can see the successful promise. However, if I try to validate again at any point, it fails for the reason "unauthorized". I don't know what that process does exactly when I run it ad hoc, because it isn't sending any requests to my back end. I'm also really confused because running `ipCookie` at any point returns a blank; I can't tell if it's storing cookies. It's driving me crazy but I'm just going to retreat to sessionStorage because I can't get it to work.

**Comment by adonespitogo** [comment]
Are you using cors? I was using cors and I read somewhere you need to include `config.headers.withCredentials` to the request. Also, you need to expose additional headers in the rack cors config to accomodate the request. This will allow the cookies to be passed from one domain to another domain. Not sure if it's your case.

**Comment by wombatdev** [comment]
So I got it working last night. I feel stupid, but the only change I made was to remove the `domain` line in the `cookieOps` object in the `$authProvider` `config` settings. In the default setup in this repo, it's set to `domain.com`, so I had been using it with it set to either `localhost:3000` (rails) or `localhost:8080` (angular). After I removed it entirely, `ipCookie()` finally began holding on to cookies, and `validateUser` works.

Sessions persist after a refresh, but too many refreshes will still invalidate the session. It doesn't seem to be a set number or time; it seems kind of random to me, so I clearly don't understand it. But it's working now, at least.
